### PR TITLE
New: [AEA-4455] - Extracts Prescription Status from Spine Response

### PR DESCRIPTION
**Pull request by JackSpagnoliNHS**
## Summary

- :sparkles: New Feature

### Details

Extracts Prescription Status from Spine Response

### CODE REVIEW

File: packages/clinicalViewLambda/tests/test-handler.test.ts [hunk 1]
```diff
@@ -7,39 +12,102 @@ import {MiddyfiedHandler} from "@middy/core"
...
-    const clinicalViewUrl = `https://live/syncservice-pds/pds`
+    mock.onPost(CLINICAL_VIEW_URL).reply(200, `<some>xml</some>`)
+
+    const event = {...MOCK_EVENT} as unknown as APIGatewayEvent
+    const context = {} as unknown as Context
+
+    await handler(event, context)
+
+    expect(mock.history.post.length).toBe(1)
+  })
```
Comment (MatthewPopat-NHS) [inline packages/clinicalViewLambda/tests/test-handler.test.ts:60]: This is redundant with the next test

File: packages/clinicalViewLambda/tests/test-handler.test.ts [hunk 2]
```diff
@@ -7,39 +12,102 @@ import {MiddyfiedHandler} from "@middy/core"
...
+
+  it("extracts prescription status from spine response", async () => {
+    mock.onPost(CLINICAL_VIEW_URL).reply(200, prescriptionFoundResponse)
+
+    const event = {...MOCK_EVENT} as unknown as APIGatewayEvent
+    const context = {} as unknown as Context
+
+    const response = await handler(event, context)
+
+    expect(mock.history.post.length).toBe(1)
```
Comment (MatthewPopat-NHS) [inline packages/clinicalViewLambda/tests/test-handler.test.ts:70]: can we use expect.toHaveBeenCalledWith instead?

**Review by wildjames — APPROVED** [review]

File: packages/clinicalViewLambda/tests/test-handler.test.ts (hunk 1 above)
Comment (JackSpagnoliNHS) [reply packages/clinicalViewLambda/tests/test-handler.test.ts]: ee4c49593e7fbf7978d19e3a07cba3456191ba70

File: packages/clinicalViewLambda/src/handler.ts [hunk 3]
```diff
@@ -2,32 +2,64 @@ import {SpineClient} from "@nhsdigital/eps-spine-client/lib/spine-client"
...
   const inboundHeaders = event.headers
+  const queryStringParameters = event.queryStringParameters ?? {}
+  const prescriptionId = event.queryStringParameters?.prescriptionId ?? ""
+
+  const clinicalViewParams = buildClinicalViewParams(inboundHeaders, queryStringParameters)
+
+  let spineResponse
+  try {
+    spineResponse = await params.spineClient.clinicalView(inboundHeaders, clinicalViewParams)
+  } catch (error) {
```
Comment (MatthewPopat-NHS) [inline packages/clinicalViewLambda/src/handler.ts:41]: why not just use the error handler middleware?

File: packages/clinicalViewLambda/tests/test-handler.test.ts (hunk 2 above)
Comment (JackSpagnoliNHS) [reply packages/clinicalViewLambda/tests/test-handler.test.ts]: 47d423eb6b49aaa2472d2d0fbe345752de90b3c3

File: packages/clinicalViewLambda/src/handler.ts (hunk 3 above)
Comment (JackSpagnoliNHS) [reply packages/clinicalViewLambda/src/handler.ts]: db075b846b1def9acd29e9b0a076fbdd261903c5

**Review by wildjames — DISMISSED** [review]

**Review by wildjames — APPROVED** [review]

**Pull request merged**